server/jetstream.go

```diff
@@ -829,32 +854,83 @@ func (a *Account) JetStreamEnabled() bool {
 	return enabled
 }
 
-// Updates accounting on in use memory and storage.
+func (jsa *jsAccount) remoteUpdateUsage(sub *subscription, c *client, subject, _ string, msg []byte) {
+	const usageSize = 16
+
+	jsa.mu.Lock()
+	s := jsa.js.srv
+	if len(msg) != usageSize {
+		jsa.mu.Unlock()
+		s.Warnf("Received remote usage update that is wrong size: %d vs %d", len(msg), usageSize)
+		return
+	}
+	var rnode string
+	if li := strings.LastIndexByte(subject, btsep); li > 0 && li < len(subject) {
+		rnode = subject[li+1:]
+	}
+	if rnode == _EMPTY_ {
+		jsa.mu.Unlock()
+		s.Warnf("Received remote usage update with no remote node")
+		return
+	}
+	var le = binary.LittleEndian
```

Comment:
Since this is a subscription on '.*', should this function ignores if the rnode is itself?

Reply:
We won't hear ourselves, no echo by default.